### DISCUSSION TITLE
Customizing BDD execution model

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [xunit/xunit#2953] [Question] [answered]
Hello, question, I have been picking up a package reference to `xunit.extensibility.execution` for my BDD [`xWellBehaved.net`](https://github.com/mwpowellhtx/xwellbehaved.net), but I wonder, can (should) we just reference `xunit`? That seemed to work for my fluent extension methods. Probably also making sense of the migration path considering [`xunit.extensions` deprecation](https://xunit.net/docs/upgrade-extensions). Thanks, cheers 🍻

**Comment by mwpowellhtx** [comment]
Which FYI, I'm not sure what happened here, something not being delivered quite right, or perhaps a legit NuGet rendering issue, do not know.

![image](https://github.com/xunit/xunit/assets/1824187/5a7fd4a8-2529-4658-80f6-ff5691b178d4)

**Reply by bradwilson** [reply]
This is a longstanding NuGet bug: https://github.com/NuGet/NuGetGallery/issues/8644#issuecomment-1601803412

That said, the next version has moved the build table to Markdown so it should render better: https://github.com/xunit/xunit/blob/6b29a2f51217d00a685b704980c6011f7f442be1/README.md?plain=1#L18-L23

**Comment by bradwilson** [comment, marked as answer]
> Hello, question, I have been picking up a package reference to xunit.extensibility.execution for my BDD [xWellBehaved.net](https://github.com/mwpowellhtx/xwellbehaved.net), but I wonder, can (should) we just reference xunit?

Definitely not. Extension projects should use the extensibility project libraries. That's why they're there. There are things you might get reference to by using `xunit` that are not designed for extensions.

> Probably also making sense of the migration path considering [xunit.extensions deprecation](https://xunit.net/docs/upgrade-extensions).

That documentation page was designed for test authors, not extensibility authors.

**Comment by mwpowellhtx** [comment]
Great, thanks! 🍻
